http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/ClusterHelper.scala

```diff
@@ -83,6 +90,8 @@ class ClusterHelper(
           } else IO.pure(Map.empty[String, String])
           clusterVPCSettings = getClusterVPCSettings(projectLabels)
 
+          resourceConstraints <- getClusterResourceContraints(cluster)
```

Comment:
if we fail to calculate constrain, we will fail cluster creation...maybe worth doing some perf test for this PR for the extra google calls?

Reply:
It's true this adds Google calls to back Leo. Hopefully they are cheap because it's just looking up machine types, not mutating anything. But good idea about perf testing.

Another way to do this is to have a static Map for memory for all Google machine types… but I didn't like that because we'd need to remember to keep it up to date.